src/gui/editors/AutomationEditor.cpp

```diff
@@ -654,6 +671,52 @@ void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
 				}
 			break;
 			}
+			case EditMode::EditTangents:
+			{
```

Comment:
The amount of nesting starting from here is a bit much (which is actually why I think I had to zoom out to view this). You could make some early returns to reduce the nesting a bit, but I won't block the PR because of this.

Reply:
As with the other case, this one is even smaller than `EditMode::Draw`. I agree the code blocks are kinda big, but maybe we should change them all at once later? It would take a little bit of rethinking for me to do it here, and the others would still be big.

Reply:
You could combine the "chain of single if-conditions" with ``&&``, and I don't think that would hurt readability or require much thinking. Other parts of the code may still be large in terms of nesting/length, but we don't have to go along with that in the interest of being consistent with the current code (I would argue that this is exactly what we shouldn't being doing as the current code doesn't set a great example of doing things most of the time).

I said I wouldn't block the PR because of this since I do realize doing extraction/early returns can sometimes just cause unintended bugs, so I wouldn't worry about it too much. I just don't like it when the code is unnecessarily verbose and nested.

Reply:
If nesting get too deep, it means you are **_highly_** motivated to separate things out into a different functions.

> I do realize doing extraction/early returns can sometimes just cause unintended bugs

Early returns don't cause bugs, they show you bugs in your thought process. :P

Reply:
> Early returns don't cause bugs, they show you bugs in your thought process.

Just didn't want to be too hard on him for this. Although if he does go ahead with early returns and extraction, I will review the code again anyways. To me, I would think it should really be a straightforward process.

Reply:
I saw a conditional on each case that could be changed to an early return, it removed one level of nesting and the conditionals were combined on the previous PR. Do you see any other changes that could improve it? Looking broadly I didn't notice anything else but I could be missing something.

Reply:
I think it's a lot better now :+1:.